lib/grakn/channel.ex

```diff
@@ -0,0 +1,125 @@
+defmodule Grakn.Channel do
+  @moduledoc false
+
+  alias Grakn.{Cache, Transaction}
+
+  @opaque t :: GRPC.Channel.t()
+
+  # every 5 min
+  @ping_rate 300_000
+
+  @spec open(String.t()) :: {:ok, t()} | {:error, any()}
+  def open(uri) do
+    GRPC.Stub.connect(uri, adapter_opts: %{http2_opts: %{keepalive: @ping_rate}})
+  end
+
+  @spec open_transaction(t(), String.t(), String.t(), String.t(), atom(), Transaction.t()) ::
+          {:ok, Grakn.Transaction.t(), String.t()} | {:error, any()}
+  def open_transaction(channel, keyspace, username, password, name, type) do
+    {session_id, cached?} = fetch_or_open_session(channel, keyspace, username, password, name)
+
+    with {:ok, tx, session_id} <- Grakn.Transaction.new(channel, session_id),
```

Comment:
Small thing, but probably it is not needed that `Grakn.Transaction.new` returns the `session_id` since it is created outside

Reply:
Good catch, changed.